docs/designs/sanitize-html.md

```diff
@@ -0,0 +1,144 @@
+# Sanitize HTML
```

Comment:
What do you think about introducing sanitization "modes" that we can refer to in our yaml schemas?
1. "Loose": the default, applies to markdown files and yaml properties strings with `contentType: markdown`.
2. "Strict": disallows html entirely. We'd start to reference this in [our schemas](https://dev.azure.com/ceapex/Engineering/_git/docs-ui/?path=%2Fpackages%2Fschemas%2Fsrc%2FModule.schema.json&version=GBmatthew%2Fsystemic-focus-colors&line=21&lineEnd=26&lineStartColumn=1&lineEndColumn=5&lineStyle=plain) via `contentType: markdown-strict` or something.

Overall we'd like to discourage use of html in content as it's a constant source of accessibility bugs, layout issues, design regressions.

Reply:
We might also need an special permissive mode for the currently archived content. I don't believe anyone wants to go back and update those pages unless they have to. The permissive mode should not be accessible to other repos though.

Reply:
How about:
1. "Loose": backwards compat... archive repos, certain page layouts like the old hub and landing pages.
2. "Standard": essentially this spec in it's current form
3. "Strict": disallow html

All modes can be activated by metadata/schema contentType.

Reply:
We can automatically detect hub/landing/archived contents to save us from changing the configs for these repos/files,